eodatasets3/prepare/sentinel_L1C_prepare.py

```diff
@@ -0,0 +1,393 @@
+
+    data_type = xmldoc.getElementsByTagName("PROCESSING_LEVEL")[0].firstChild.data
+    datastrip_id = xmldoc.getElementsByTagName("PRODUCT_URI")[0].firstChild.data
+    product_type = xmldoc.getElementsByTagName("PRODUCT_TYPE")[0].firstChild.data
+    platform = xmldoc.getElementsByTagName("SPACECRAFT_NAME")[0].firstChild.data
+    orbit = xmldoc.getElementsByTagName("SENSING_ORBIT_NUMBER")[0].firstChild.data
+    orbit_direction = xmldoc.getElementsByTagName("SENSING_ORBIT_DIRECTION")[
+        0
+    ].firstChild.data
+    datatake_type = xmldoc.getElementsByTagName("DATATAKE_TYPE")[0].firstChild.data
+    processing_datetime = xmldoc.getElementsByTagName("GENERATION_TIME")[
+        0
+    ].firstChild.data
+    processing_baseline = xmldoc.getElementsByTagName("PROCESSING_BASELINE")[
+        0
+    ].firstChild.data
+    datetime = xmldoc.getElementsByTagName("PRODUCT_START_TIME")[0].firstChild.data
+    cloud_cover = xmldoc.getElementsByTagName("Cloud_Coverage_Assessment")[
+        0
+    ].firstChild.data
+    region_code = datastrip_id.split("_")[5][1:]
+
+    return {
+        "data_type": data_type,
+        "datastrip_id": datastrip_id,
+        "product_type": product_type,
+        "platform": platform,
+        "orbit": orbit,
+        "orbit_direction": orbit_direction,
+        "datatake_type": datatake_type,
+        "processing_datetime": processing_datetime,
+        "processing_baseline": processing_baseline,
+        "datetime": datetime,
+        "cloud_cover": cloud_cover,
+        "region_code": region_code,
+    }
+
+
+def process_format_correctness(
+    format_correctness_path,
+    zip_object,
+):
+    if zip_object is not None:
+        xmldoc = minidom.parseString(zip_object.read(format_correctness_path))
+
+        source_system = xmldoc.getElementsByTagName("System")[0].firstChild.data
+        software_version = xmldoc.getElementsByTagName("Creator_Version")[
+            0
+        ].firstChild.data
+
+        return {
+            "source_system": source_system,
+            "software_version": software_version,
+        }
+    else:
+        xmldoc = minidom.parse(format_correctness_path)
+
+        source_system = xmldoc.getElementsByTagName("System")[0].firstChild.data
+        creator_version = xmldoc.getElementsByTagName("Creator_Version")[
+            0
+        ].firstChild.data
+        creation_date = xmldoc.getElementsByTagName("Creation_Date")[0].firstChild.data
+        datastrip_metadata = xmldoc.getElementsByTagName("File_Name")[0].firstChild.data
+
+        return {
+            "source_system": source_system,
+            "creator_version": creator_version,
+            "creation_date": creation_date,
+            "datastrip_metadata": datastrip_metadata,
+        }
+
+
+def prepare_and_write(
+    dataset: Path,
+    dataset_document: Path,
+) -> Tuple[uuid.UUID, Path]:
+    # Process esa dataset
+    if "zip" in str(dataset):
```

Comment:
I'm a bit iffy about using 'zip' as the way to distinguish the type of dataset (ESA vs Synergise). From experience, people often try to run these things with unzipped folders or otherwise, and it's going to quietly try to parse all the metadata differently depending on whether someone has done that.

Is there a file inside that we could use to detect? Or a more specific part of the filename?